use nginx variable mapping to provide log keys with proper values

alexanderGalushka opened the pull request:
We've observed dashes logged as upstream response time as well as edgescape and ua. This PR addressed it, setting edgescape to none if it's empty, setting time to 0.000 if empty and ua to unknown if empty.

Jira ticket https://jira.redbullmediahouse.com/browse/RBTV-26063

alexanderGalushka commented:
it's up on staging and does proper logging `measure#nginx.service=0.000 request_id=cdc32b39-5cf2-4eaa-9041-5a0526cb5cce path=/v3/products upstream_response_time=0.000 country=us edgescape=none user_agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36" status=200 method=GET`

stran12 commented on config/nginx.conf.erb:
```diff
@@ -17,7 +17,30 @@ http {
+    "~country_code=([A-Z]{2})"               "$1";
+  }
+
+  log_format l2met 'measure#nginx.service=$request_time request_id=$http_x_request_id path=$request_uri upstream_response_time=$urt country=$country edgescape=$esc user_agent="$ua" status=$status method=$request_method';
```
why is `"$ua"` in quotes and everything else is not?

base698 commented:
Ua has spaces and doesn't parse correctly.

On Aug 21, 2017 9:22 AM, "Stephen Tran" <notifications@github.com> wrote:

> *@stran12* commented on this pull request.
> ------------------------------
>
> In config/nginx.conf.erb
> <https://github.com/RBNA/nginx-buildpack/pull/4#discussion_r134275597>:
>
> > +    "-"                   "unknown";
> +    ""                    "unknown";
> +  }
> +
> +  map $http_X_Akamai_Edgescape $esc {
> +    default               $http_X_Akamai_Edgescape;
> +    "-"                   "none";
> +    ""                    "none";
> +  }
> +
> +  map $http_X_Akamai_Edgescape $country {
> +    default               "us";
> +    "~country_code=([A-Z]{2})"               "$1";
> +  }
> +
> +  log_format l2met 'measure#nginx.service=$request_time request_id=$http_x_request_id path=$request_uri upstream_response_time=$urt country=$country edgescape=$esc user_agent="$ua" status=$status method=$request_method';
>
> why is "$ua" in quotes and everything else is not?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/RBNA/nginx-buildpack/pull/4#pullrequestreview-57548633>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAMqmmCHAgJjmUS5dU3l171W062rhg7aks5saa7NgaJpZM4O8Ges>
> .
>

stran12 commented:
Should we then use quotes for everything just in case?

stran12 commented:
#beconsistent

base698 commented:
If it's not a number

On Aug 21, 2017 9:33 AM, "Stephen Tran" <notifications@github.com> wrote:

> #beconsistent
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/RBNA/nginx-buildpack/pull/4#issuecomment-323793002>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAMqmgcs9mOwI6JmuMFKYr8fEfeboYXsks5sabFlgaJpZM4O8Ges>
> .
>

alexanderGalushka commented:
@stran12 @base698 I'll have quotes for the strings only then

alexanderGalushka commented:
@stran12 @base698 addressed the comment, the logs will look like the following `measure#nginx.service=0.176 request_id="ea010c9f-2c56-4b80-9093-0c31c7620599" path="/v3/products" upstream_response_time=0.176 country="us" edgescape="none" user_agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36" status=200 method="GET"`
